src/coreclr/vm/stubgen.cpp

```diff
@@ -819,10 +834,10 @@ size_t ILStubLinker::Link(UINT* puMaxStack)
     }
 
 #ifdef _DEBUG
-    if (fStackUnderflow)
+    if (fStackUnderflow || this->GetNumEHClauses() != 0)
     {
         LogILStub(CORJIT_FLAGS());
-        CONSISTENCY_CHECK_MSG(false, "IL stack underflow! -- see logging output");
+//        CONSISTENCY_CHECK_MSG(false, "IL stack underflow! -- see logging output");
     }
 #endif // _DEBUG
```

Comment:
We should undo this change, I think it is just an old debug change accidentally committed

Reply:
This has been undone.